samba: Add documentation for addon_configs (#3322)
home-assistant/addons: samba/DOCS.md

Directory | Description
-- | --
`addons` | This is for your local add-ons.

> `addon_configs` | This is for your the configuration files or your add-ons.

Reviewer: ```suggestion
`addon_configs` | This is for your add-on's configuration files.
```

Author: That implies singular, which it isn't?

Reviewer: Right, I think this should do it.

```suggestion
`addon_configs` | This is for your add-ons' configuration files.
```

Reviewer: Or I guess the of form which might be nicer? Either way works for me.

```suggestion
`addon_configs` | This is for the configuration files of your add-ons.
```
